include/glow/Base/Type.h

```diff
@@ -390,6 +390,30 @@ struct Type final {
     return true;
   }
 
+  /// \returns a hash value for this Type. Hashes for Ty1 and Ty2 are equal iF
+  /// Ty1.isEqual(Ty2).
+  size_t hash() const {
```

Comment:
You can use llvm::hash. This is what we do in other places in the code.

Reply:
Unless I'm misreading we define the hash value of Tensor to be literally its size right now though...

https://github.com/pytorch/glow/blob/master/lib/Graph/Nodes.cpp#L1325

Reply:
@nadavrot I might be missing what you meant here, what do you suggest?

Reply:
LLVM has utilities for computing the hash value of things. It can merge hashes for you as well. This means that you don't need to implement the shifts and xors yourself. Just call into LLVM's hash method and it will digest things for you.